src/util/run.cpp

```diff
@@ -87,10 +87,34 @@ int run(
   const std::string &std_error)
 {
   #ifdef _WIN32
-  // we don't support stdin/stdout/stderr redirection on Windows
-  PRECONDITION(std_input.empty());
-  PRECONDITION(std_output.empty());
-  PRECONDITION(std_error.empty());
+  // we use the cmd.exe shell to do stdin/stdout/stderr redirection on Windows
+  if(!std_input.empty() || !std_output.empty() || !std_error.empty())
+  {
+    std::vector<std::string> new_argv = argv;
+    new_argv.insert(new_argv.begin(), "cmd.exe");
+    new_argv.insert(new_argv.begin() + 1, "/c");
+
+    if(!std_input.empty())
+    {
+      new_argv.push_back("<");
+      new_argv.push_back(std_input);
+    }
+
+    if(!std_output.empty())
+    {
+      new_argv.push_back(">");
+      new_argv.push_back(std_output);
+    }
+
+    if(!std_error.empty())
+    {
+      new_argv.push_back("2>");
+      new_argv.push_back(std_error);
+    }
+
+    // this is recursive
+    return run("cl.exe", new_argv, "", "", "");
```

Comment:
So there's still this invocation of `cl.exe` left to be resolved.

Reply:
Indeed, now fixed.